triggers/service/src/test/scala/com/digitalasset/daml/lf/engine/trigger/ServiceTest.scala

```diff
@@ -213,45 +223,77 @@ class ServiceTest extends AsyncFlatSpec with Eventually with Matchers with Postg
   it should "start multiple triggers and list them by party" in withHttpService(Some(dar)) {
     (uri: Uri, client) =>
       for {
-        // no triggers running initially
         resp <- listTriggers(uri, "Alice")
         result <- parseTriggerIds(resp)
         _ <- result should equal(Vector())
         // start trigger for Alice
         resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Alice")
         aliceTrigger <- parseTriggerId(resp)
-        resp <- listTriggers(uri, "Alice")
-        result <- parseTriggerIds(resp)
-        _ <- result should equal(Vector(aliceTrigger))
-        // start trigger for Bob
+        _ <- Future {
+          eventually {
+            val r = Await.result(for {
+              resp <- listTriggers(uri, "Alice")
+              result <- parseTriggerIds(resp)
+            } yield result, Duration.Inf)
+            assert(r == Vector(aliceTrigger))
+          }
+        }
         resp <- startTrigger(uri, s"$testPkgId:TestTrigger:trigger", "Bob")
         bobTrigger1 <- parseTriggerId(resp)
-        resp <- listTriggers(uri, "Bob")
-        result <- parseTriggerIds(resp)
-        _ <- result should equal(Vector(bobTrigger1))
-        // start another trigger for Bob
+        _ <- Future {
+          eventually {
+            val r = Await.result(for {
```

Comment:
This seems like a helper function `assertTriggerIds(uri, party, expectedTriggerIds)` would be very useful.

Reply:
Indeed!

Reply:
PR https://github.com/digital-asset/daml/pull/6147